larq_zoo/sota/quicknet.py

```diff
@@ -30,42 +31,49 @@ def LCEFirstLayer(filters: int, x: tf.Tensor) -> tf.Tensor:
     return tf.keras.layers.BatchNormalization(momentum=0.9, epsilon=1e-5)(x)
 
 
+def squeeze_and_excite(inp: tf.Tensor, filters: int, r: int = 16):
+    """Squeeze and Excite as per [Squeeze-and-Excitation Networks](https://arxiv.org/abs/1709.01507)"""
+    C = inp.get_shape().as_list()[-1]
+
+    out = utils.global_pool(inp)
+    out = tf.keras.layers.Dense(
+        C // r,
+        activation="relu",
+        kernel_initializer="he_normal",
+        use_bias=False,
+        kernel_regularizer=tf.keras.regularizers.l2(1e-5),
+    )(out)
+
+    out = tf.keras.layers.Dense(
+        filters,
+        activation="sigmoid",
+        kernel_initializer="he_normal",
+        use_bias=False,
+        kernel_regularizer=tf.keras.regularizers.l2(1e-5),
+    )(out)
+
+    return tf.reshape(out, [-1, 1, 1, filters])
+
+
 @factory
-class QuickNetFactory(ModelFactory):
-    """Quicknet - A model designed for fast inference using [Larq Compute Engine](https://github.com/larq/compute-engine)"""
+class QuickNetBaseFactory(ModelFactory):
 
-    num_layers: int = Field(15)
+    spec: Tuple[Sequence[int], Sequence[int], Sequence[bool]] = Field(None)
```

Comment:
The last spec for the SE, although useful here, means that as we transition to being more precise with where we place SE, that this will need to be modified

Reply:
Yes that's a good point, but I'm not sure if we can/should anticipate this now. Do you have a concrete proposal of how to implement this? Otherwise I think it's fine to modify this again when the time comes.